hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -1061,8 +1061,14 @@ public static SingularAssociationAttributeMapping buildSingularAssociationAttrib
 						sessionFactory
 				);
 
-		final FetchTiming fetchTiming = FetchStrategyHelper.determineFetchTiming( fetchStyle, type, sessionFactory );
+		final FetchTiming fetchTiming;
 
+		if ( !( (ToOne) bootProperty.getValue() ).isLazy() ) {
```

Comment:
I think we cannot take it for granted that the downcast is always valid. Seems `ToOne` is only one child class type for `AssociationType`. We might well fix the issue by `instanceof` checking first but it would be great to fix loophole for other association type.

Reply:
Yes that is always a safe bet *now*.  To be determined what happens when we implement ANY support